writer.go

```diff
@@ -49,394 +52,524 @@ func (w *Writer) Flush() error {
 	return w.w.Flush()
 }
 
-func (w *Writer) writeFEDWireMessage(file *File) error {
+// options
+//  first bool : has variable length
+//  second bool : has not new line
+func (w *Writer) writeFEDWireMessage(file *File, options ...bool) error {
+
+	newLine := "\n"
+	if hasNotNewLine(options) {
```

Comment:
With the suggestion above, all of these become `if w.noNewlines { ... }` instead of checking "the second boolean flag".

What might be even better then a boolean flag here is to simply include a `newlineChar` field on the `Writer` struct. Then you could have an options function like:
```go
func NewlineCharacter(newline string) OptionFunc {
	return func(w *Writer) {
		w.newlineChar = newline
	}
}

// NewWriter returns a new Writer that writes to w.
func NewWriter(w io.Writer, opts ...OptionFunc) *Writer {
	writer := &Writer{
		w: bufio.NewWriter(w),
		newlineChar: "\n",
	}

	for _, opt := range opts {
		opt(writer)
	}

	return writer
}
```

This way it would default to `"\n"` like you've currently implemented, but you wouldn't have to repeat this code block anywhere else.